clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -760,18 +760,43 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
+            Map<String, String> metricTags = new HashMap<>();
+            metricTags.put("topic", "topic-" + topic.replace(".", "_"));
```

Comment:
I think it was my suggestion to add it. I was going off the pattern for tagged node metrics in Selector.